RecoVertex/Configuration/python/RecoVertex_cff.py

```diff
@@ -36,33 +36,78 @@
 from SimTracker.TrackAssociation.trackTimeValueMapProducer_cfi import trackTimeValueMapProducer
+from CommonTools.RecoAlgos.TOFPIDProducer_cfi import TOFPIDProducer
 _phase2_tktiming_vertexrecoTask = cms.Task( vertexrecoTask.copy() ,
                                             tpClusterProducer ,
                                             quickTrackAssociatorByHits ,
                                             trackTimeValueMapProducer ,
-                                            trackWithVertexRefSelectorBeforeSorting4D ,
-                                            trackRefsForJetsBeforeSorting4D ,
-                                            unsortedOfflinePrimaryVertices4D ,
-                                            offlinePrimaryVertices4D ,
-                                            offlinePrimaryVertices4DWithBS 
+                                            unsortedOfflinePrimaryVertices4DnoPID ,
+                                            trackWithVertexRefSelectorBeforeSorting4DnoPID ,
+                                            trackRefsForJetsBeforeSorting4DnoPID ,
+                                            offlinePrimaryVertices4DnoPID ,
+                                            offlinePrimaryVertices4DnoPIDWithBS,
+                                            TOFPIDProducer,
+                                            unsortedOfflinePrimaryVertices4Dfastsim,
+                                            trackWithVertexRefSelectorBeforeSorting4Dfastsim ,
+                                            trackRefsForJetsBeforeSorting4Dfastsim ,
+                                            offlinePrimaryVertices4Dfastsim,
+                                            offlinePrimaryVertices4DfastsimWithBS,
+                                            unsortedOfflinePrimaryVertices3D,
+                                            trackWithVertexRefSelectorBeforeSorting3D ,
+                                            trackRefsForJetsBeforeSorting3D,
+                                            offlinePrimaryVertices3D,
+                                            offlinePrimaryVertices3DWithBS,
                                             )
 
 from Configuration.Eras.Modifier_phase2_timing_cff import phase2_timing
 phase2_timing.toReplaceWith(vertexrecoTask, _phase2_tktiming_vertexrecoTask)
-
+phase2_timing.toReplaceWith(unsortedOfflinePrimaryVertices, unsortedOfflinePrimaryVertices4D)
+phase2_timing.toReplaceWith(offlinePrimaryVertices, offlinePrimaryVertices4D)
+phase2_timing.toReplaceWith(offlinePrimaryVerticesWithBS, offlinePrimaryVertices4DWithBS)
+phase2_timing.toModify(offlinePrimaryVertices, vertices = cms.InputTag("unsortedOfflinePrimaryVertices"), particles = cms.InputTag("trackRefsForJetsBeforeSorting"))
+phase2_timing.toModify(offlinePrimaryVerticesWithBS, vertices = cms.InputTag("unsortedOfflinePrimaryVertices","WithBS"), particles = cms.InputTag("trackRefsForJetsBeforeSorting"))
```

Comment:
please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.